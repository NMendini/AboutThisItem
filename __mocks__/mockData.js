module.exports = {
  details: {
    "highlights": [
        "MAGNI REPELLAT AUT: Facere odio fugiat nesciunt voluptatem rerum omnis accusantium aut adipisci.",
        "ATQUE CONSEQUATUR ET: Consequatur error fugit sint possimus.\nVel id cum dolor.",
        "IURE AT SOLUTA: Porro dolore libero nisi.\nVel rerum non aliquid ut.",
        "QUOS EUM VEL: Aspernatur excepturi qui eaque consequatur voluptatem id consectetur officia dicta.\nSit aut aliquid sunt dignissimos id est perferendis deleniti deleniti.\nFuga vel quo qui odit animi et id.\nQuos rerum corporis quis pariatur nulla sed id.\nUt consectetur temporibus.",
        "EAQUE VOLUPTATEM PARIATUR: Sit repellat rem veritatis nihil quaerat eum quia quis.",
        "ATQUE EA AUT: Corporis quas magnam adipisci consequatur dolore et ipsum et quae.\nSed hic omnis ducimus rerum odit possimus.\nTempore adipisci eos quod rem.\nEa quos architecto corporis.\nDolore in rem quisquam ut quia."
    ],
    "specifications": {
        "Item Number": "1",
        "Frozen": "aut dicta eligendi",
        "Rubber": "et odit voluptas",
        "Fresh": "quas qui iste",
        "Metal": "nostrum officia sunt",
        "Wooden": "dolor fugit nihil",
        "Granite": "enim architecto at",
        "Cotton": "sint repellendus velit",
        "Plastic": "omnis ratione iure",
        "Steel": "et quisquam animi",
        "Height": "17.25 inches",
        "Width": "24.07 inches",
        "Length": "27.23 inches",
        "Weight": "5.48 pounds"
    },
    "description": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support. Labore quasi praesentium reiciendis et expedita magnam ut. Aut ex quod voluptatem dolorem id et aut fuga et. Sit distinctio magni sit suscipit molestias quae et. Vel quo fugiat blanditiis placeat. Consequuntur itaque aliquam sequi ut.\n \rIusto quas corrupti tempora. Animi neque neque ut blanditiis aliquam eveniet rem. Sed officia reiciendis nulla vel aliquam eum iste tenetur.\n \rQuia debitis libero rem ad qui officia alias ut optio. Vel non qui iure. Nobis doloribus voluptatem quisquam omnis. Et rerum quae sint atque laborum tenetur excepturi ullam. Officiis et aut ea."
  },
  questions: [
    {
        "question_id": 1,
        "user_name": "Dedrick",
        "question": "Aut eum aut laboriosam id laborum vel omnis voluptas?",
        "created_at": "2020-04-02T17:51:57.000Z",
        "product_id": 91
    },
    {
        "question_id": 2,
        "user_name": "Freda",
        "question": "Ea quisquam atque molestiae non qui accusantium?",
        "created_at": "2020-05-18T20:14:35.000Z",
        "product_id": 91
    }
  ],
  answers: [
    {
        "answer_id": 1,
        "user_name": "Nichole",
        "question_id": 1,
        "answer": "Neque officiis quod quo perferendis voluptatum aut sed. Aut in pariatur tempora et animi saepe non. Iure neque quia qui qui qui excepturi. Quod vitae pariatur nisi neque quidem voluptates maxime qui.",
        "created_at": "2020-01-15T19:45:38.000Z",
        "helpful": 1,
        "not_helpful": 12,
        "product_id": 1
    },
    {
        "answer_id": 2,
        "user_name": "Arianna",
        "question_id": 1,
        "answer": "Qui sit vel aut molestiae. Expedita quia ipsam. Soluta accusantium aspernatur consequatur quibusdam magnam quas. Voluptas fuga sapiente nam distinctio deserunt fuga atque.",
        "created_at": "2020-05-03T22:34:22.000Z",
        "helpful": 19,
        "not_helpful": 27,
        "product_id": 1
    },
    {
      "answer_id": 3,
      "user_name": "Joshua",
      "question_id": 2,
      "answer": "Soluta optio deserunt ipsam aut hic nostrum quaerat. Voluptatem accusamus harum.",
      "created_at": "2020-03-09T08:11:04.000Z",
      "helpful": 16,
      "not_helpful": 35,
      "product_id": 1
    },
    {
        "answer_id": 4,
        "user_name": "Hubert",
        "question_id": 2,
        "answer": "Nisi distinctio praesentium. Qui maxime qui a laborum eum quia sed vero quae. Cupiditate voluptatem aut culpa dolores omnis enim voluptas recusandae. Assumenda et voluptatem sed unde impedit autem cupiditate cum.",
        "created_at": "2020-03-12T15:16:48.000Z",
        "helpful": 37,
        "not_helpful": 39,
        "product_id": 1
    }
  ]
}