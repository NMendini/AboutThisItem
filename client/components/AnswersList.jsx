import React from 'react';
import AnswersListEntry from './AnswersListEntry.jsx';
import styles from '../assets/style.css';

const AnswersList = function(props) {
  return (
    <div className={styles.containerAnswer}>
      {
        props.answers.map((item, i) => {
          if (item['question_id'] === props.questionId) {
            return <AnswersListEntry
              answer={item}
              key={i}
            />;
          }
        })
      }
    </div>
  );
};

export default AnswersList;