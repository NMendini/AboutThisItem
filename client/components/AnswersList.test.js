/**
 * @jest-environment jsdom
 */
import 'babel-polyfill';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import mockAxios from 'jest-mock-axios';
import { details, questions, answers }from '../../__mocks__/mockData.js';

// COMPONENTS
import AnswersList from './AnswersList.jsx';

let component;
let props;

beforeAll(async () => {
  props = {
    answers: answers,
    questionId: 1
  }

  component = mount(<AnswersList {...props}/>);


});

afterAll(async () => {
  await component.unmount();
});

describe('AnswersList Component', () => {
  it('should render AnswersList component correctly', async () => {
      expect(component).toMatchSnapshot();
  });
});