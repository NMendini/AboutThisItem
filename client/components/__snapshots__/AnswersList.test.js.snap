// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswersList Component should render AnswersList component correctly 1`] = `
<AnswersList
  answers={
    Array [
      Object {
        "answer": "Neque officiis quod quo perferendis voluptatum aut sed. Aut in pariatur tempora et animi saepe non. Iure neque quia qui qui qui excepturi. Quod vitae pariatur nisi neque quidem voluptates maxime qui.",
        "answer_id": 1,
        "created_at": "2020-01-15T19:45:38.000Z",
        "helpful": 1,
        "not_helpful": 12,
        "product_id": 1,
        "question_id": 1,
        "user_name": "Nichole",
      },
      Object {
        "answer": "Qui sit vel aut molestiae. Expedita quia ipsam. Soluta accusantium aspernatur consequatur quibusdam magnam quas. Voluptas fuga sapiente nam distinctio deserunt fuga atque.",
        "answer_id": 2,
        "created_at": "2020-05-03T22:34:22.000Z",
        "helpful": 19,
        "not_helpful": 27,
        "product_id": 1,
        "question_id": 1,
        "user_name": "Arianna",
      },
      Object {
        "answer": "Soluta optio deserunt ipsam aut hic nostrum quaerat. Voluptatem accusamus harum.",
        "answer_id": 3,
        "created_at": "2020-03-09T08:11:04.000Z",
        "helpful": 16,
        "not_helpful": 35,
        "product_id": 1,
        "question_id": 2,
        "user_name": "Joshua",
      },
      Object {
        "answer": "Nisi distinctio praesentium. Qui maxime qui a laborum eum quia sed vero quae. Cupiditate voluptatem aut culpa dolores omnis enim voluptas recusandae. Assumenda et voluptatem sed unde impedit autem cupiditate cum.",
        "answer_id": 4,
        "created_at": "2020-03-12T15:16:48.000Z",
        "helpful": 37,
        "not_helpful": 39,
        "product_id": 1,
        "question_id": 2,
        "user_name": "Hubert",
      },
    ]
  }
  questionId={1}
>
  <div
    className="containerAnswer"
  >
    <AnswersListEntry
      answer={
        Object {
          "answer": "Neque officiis quod quo perferendis voluptatum aut sed. Aut in pariatur tempora et animi saepe non. Iure neque quia qui qui qui excepturi. Quod vitae pariatur nisi neque quidem voluptates maxime qui.",
          "answer_id": 1,
          "created_at": "2020-01-15T19:45:38.000Z",
          "helpful": 1,
          "not_helpful": 12,
          "product_id": 1,
          "question_id": 1,
          "user_name": "Nichole",
        }
      }
      key="0"
    >
      <div>
        <div
          className="answerRow"
        >
          <span
            className="answerBold"
          >
            A: 
          </span>
          <span>
            Neque officiis quod quo perferendis voluptatum aut sed. Aut in pariatur tempora et animi saepe non. Iure neque quia qui qui qui excepturi. Quod vitae pariatur nisi neque quidem voluptates maxime qui.
          </span>
        </div>
        <div
          className="answerRowName"
        >
          <span>
            Nichole
          </span>
           — 
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element={null}
            filter={[Function]}
            fromNow={true}
            interval={60000}
            local={false}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            <time
              dateTime={"2020-01-15T19:45:38.000Z"}
            >
              10 months ago
            </time>
          </t>
           
          <span
            className="helpful answerClick"
          >
             Helpful(
            1
            ) 
          </span>
          <span
            className="not-helpful answerClick"
          >
             Not helpful(
            12
            ) 
          </span>
          <span
            className="report-answer answerClick"
          >
             Report 
          </span>
        </div>
      </div>
    </AnswersListEntry>
    <AnswersListEntry
      answer={
        Object {
          "answer": "Qui sit vel aut molestiae. Expedita quia ipsam. Soluta accusantium aspernatur consequatur quibusdam magnam quas. Voluptas fuga sapiente nam distinctio deserunt fuga atque.",
          "answer_id": 2,
          "created_at": "2020-05-03T22:34:22.000Z",
          "helpful": 19,
          "not_helpful": 27,
          "product_id": 1,
          "question_id": 1,
          "user_name": "Arianna",
        }
      }
      key="1"
    >
      <div>
        <div
          className="answerRow"
        >
          <span
            className="answerBold"
          >
            A: 
          </span>
          <span>
            Qui sit vel aut molestiae. Expedita quia ipsam. Soluta accusantium aspernatur consequatur quibusdam magnam quas. Voluptas fuga sapiente nam distinctio deserunt fuga atque.
          </span>
        </div>
        <div
          className="answerRowName"
        >
          <span>
            Arianna
          </span>
           — 
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element={null}
            filter={[Function]}
            fromNow={true}
            interval={60000}
            local={false}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            <time
              dateTime={"2020-05-03T22:34:22.000Z"}
            >
              7 months ago
            </time>
          </t>
           
          <span
            className="helpful answerClick"
          >
             Helpful(
            19
            ) 
          </span>
          <span
            className="not-helpful answerClick"
          >
             Not helpful(
            27
            ) 
          </span>
          <span
            className="report-answer answerClick"
          >
             Report 
          </span>
        </div>
      </div>
    </AnswersListEntry>
  </div>
</AnswersList>
`;
